### PR TITLE
rbd-mirror: image deletions should be handled by assigned instance

### PR DESCRIPTION
**Pull request by dillaman**

### CODE REVIEW

File: src/tools/rbd_mirror/image_replayer/BootstrapRequest.h [hunk 1]
```diff
@@ -175,9 +175,6 @@ class BootstrapRequest : public BaseRequest {
 
   bufferlist m_out_bl;
 
-  void get_local_image_id();
-  void handle_get_local_image_id(int r);
```
Comment (trociny) [inline src/tools/rbd_mirror/image_replayer/BootstrapRequest.h]: need to update the state diagram

Comment (dillaman) [inline src/tools/rbd_mirror/image_replayer/BootstrapRequest.h]: ack

File: src/tools/rbd_mirror/ImageReplayer.cc [hunk 2]
```diff
@@ -335,7 +336,8 @@ void ImageReplayer<I>::add_remote_image(const std::string &mirror_uuid,
 
 template <typename I>
 void ImageReplayer<I>::remove_remote_image(const std::string &mirror_uuid,
-                                           const std::string &image_id) {
+                                           const std::string &image_id,
+					   bool schedule_delete) {
```
Comment (trociny) [inline src/tools/rbd_mirror/ImageReplayer.cc]: nit: you might want to use spaces vs tabs here

File: src/tools/rbd_mirror/ImageReplayer.h [hunk 3]
```diff
@@ -109,7 +109,8 @@ class ImageReplayer {
                         const std::string &remote_image_id,
                         librados::IoCtx &remote_io_ctx);
   void remove_remote_image(const std::string &remote_mirror_uuid,
-                           const std::string &remote_image_id);
+                           const std::string &remote_image_id,
+			   bool schedule_delete);
```
Comment (trociny) [inline src/tools/rbd_mirror/ImageReplayer.h]: tabs vs spaces

File: src/tools/rbd_mirror/InstanceReplayer.cc [hunk 4]
```diff
@@ -190,11 +190,13 @@ void InstanceReplayer<I>::release_image(const std::string &global_image_id,
   }
 
   auto image_replayer = it->second;
-
-  image_replayer->remove_remote_image(peer_mirror_uuid, peer_image_id);
+  if (!peer_mirror_uuid.empty()) {
+    image_replayer->remove_remote_image(peer_mirror_uuid, peer_image_id,
+					schedule_delete);
```
Comment (trociny) [inline src/tools/rbd_mirror/InstanceReplayer.cc]: tabs

**Pull request merged**